File: benchmarks/fibonacci/bench.py
import time
def fib_count():
    U64_MAX = (1 << 64) - 1
    a, b, count = 0, 1, 0
    while True:
        nxt = a + b
        if nxt > U64_MAX: break
        a, b = b, nxt; count += 1
        if b == U64_MAX: break
    return count
t = time.perf_counter()
n = fib_count()
print(f"Fibonacci numbers under u64: {n} ({(time.perf_counter()-t)*1000:.3f}ms)")

File: benchmarks/fibonacci/bench.rs
use std::time::Instant;
fn fib_count() -> u32 {
    let (mut a, mut b): (u64, u64) = (0, 1);
    let mut count = 0u32;
    loop {
        let next = a.wrapping_add(b);
        if next < b { break; }
        a = b; b = next; count += 1;
        if b == u64::MAX { break; }
    }
    count
}
fn main() {
    let t = Instant::now();
    let n = fib_count();
    println!("Fibonacci numbers under u64: {} ({}ms)", n, t.elapsed().as_millis());
}
